misc/package/Data Files/MWSE/core/lib/dependencyManagement/DependencyManager.lua

```diff
@@ -74,8 +78,53 @@ end
 
 
 ---Check if all dependencies are met
----@return boolean #returns true if all dependencies passed, false if any failed.
+---@return boolean passed #returns true if all dependencies passed, false if any failed.
 function DependencyManager:checkDependencies()
+    if not mwse.getConfig("EnableDependencyChecks") then
+        return true
+    end
+
+    -- Don't check dependencies if a mod't plugin is not active.
+    -- Log if there is metadata.toml pointing to a partially installed/uninstalled mod.
+    local plugin = self.metadata.package.plugin
+    local pluginExists = plugin and util.pluginExists(plugin)
+    local pluginDisabled = pluginExists and not tes3.isModActive(plugin)
+
+    local luaMod = self.metadata.tools and self.metadata.tools.mwse and self.metadata.tools.mwse["lua-mod"]
+    local luaModExists = luaMod and util.luaModExists(luaMod)
+
+    local luaOnly = luaMod and not plugin or false
+    local pluginOnly = plugin and not luaMod or false
+    local luaPlugin = luaMod and plugin and true or false
+
+    local uncomplete = (luaOnly and not luaModExists) or
+                       (pluginOnly and not pluginExists) or
+                       (luaPlugin and not (luaModExists and pluginExists))
+
+    if uncomplete then
+        self.logger:warn("Metadata file (%s) found pointing to missing mod files:",
+            self.metadata.package.name
+        )
+        if not pluginExists then
+            self.logger:warn("Plugin: \"%s\".", plugin)
+        end
+        if not luaModExists then
+            self.logger:warn("MWSE lua-mod: \"%s\".", luaMod)
+        end
+        self.logger:warn("No dependency checking will be performed.")
+        self.logger:warn("This can result from uncomplete mod installation or uninstallation.")
+
+        return true
+    end
+
+    if pluginDisabled then
+        if self.showFailureMessage then
+            self.logger:info("Plugin \"%s\" is not active, skipping dependency check.", plugin)
+        end
```

Comment:
showFailureMessage is for determining if the user is notified in-game, it shouldn't be used for filtering logs.